src/mesh/mesh_base.C

```diff
@@ -218,6 +222,43 @@ void MeshBase::set_elem_dimensions(const std::set<unsigned char> & elem_dims)
   _elem_dims = elem_dims;
 }
 
+
+
+void MeshBase::add_elemset_code(dof_id_type code, const std::set<elemset_id_type> & id_set)
```

Comment:
Should we pass id_set by value, so we can do move construction and avoid a copy in some cases?

Reply:
The current "two maps" approach uses `id_set` twice:
```
void MeshBase::add_elemset_code(dof_id_type code, const std::set<elemset_id_type> & id_set)
{
  _elemset_codes.emplace(code, id_set);
  _elemset_codes_inverse_map.emplace(id_set, code);
}
```
so I think you'd need to `std::move()` it in the first call, leaving it invalid for the second call. Moving to a single map approach of some kind would help with this though.

Reply:
Even with the two-maps design we still would go from two set copy constructions down to one.  But yeah, we'd have to make one of those maps into a map-to-pointers to go from one down to zero.

Reply:
I updated `add_elemset_code` to use pass-by-value semantics.